lib/cloudimage/uri.rb

```diff
@@ -32,14 +34,22 @@ def initialize(base_url, path)
       alias_method from, to
     end
 
-    def to_url(extra_params = {})
-      url_params = params.merge(extra_params)
-      uri.query_values = url_params if url_params.any?
+    def to_url(**extra_params)
+      set_uri_params(extra_params)
       uri.to_s
     end
 
     private
 
+    def set_uri_params(**extra_params)
+      url_params = params.merge(extra_params)
+      uri.query_values = url_params if url_params.any?
+    end
+
+    def signature
+      Digest::SHA1.hexdigest(salt + uri.request_uri.sub('/v7', ''))
```

Comment:
No totally understand the remove of the /v7 - Can you explain a bit more?

Reply:
I think I understood, the full URL contains the V7 and you remove it if it's in the URL - may be in this case use the variable API_VERSION

Reply:
Hi @pfranck this is still a WIP but yes, given a URL `https://demosign.cloudimg.io/v7/sample.li/boat.jpg?w=200` we need to use salt and `sample.li/boat.jpg?w=200` to compute SHA1, otherwise the signature is incorrect as evaluated by the API. And certainly, the version string will be generated from the constant in the final PR.